### PR TITLE
pet repo completed, slight changes to api and db folders

### PR DESCRIPTION
**Pull request by Joshdike**

### CODE REVIEW

File: api/v1/gostream.proto [hunk 1]
```diff
@@ -68,7 +68,8 @@ message CreateUserResponse {
 }
 
 message UpdateUserRequest {
-  User user = 1;
+  uint64 user_id = 1;
+  User user = 2;
```
Comment (lordvidex) [inline api/v1/gostream.proto:72]: Following the `UpdatePetRequest` format, this message should also be like 

```go
message UpdateUserRequest {
  uint64 user_id = 1;
  User new_data = 2;
  }
```

You can also add http options to the update service like:
```proto
option (google.api.http) = {
      put: "/user/{user_id}"
      body: "*"
    };
```

File: internal/db/pg/pets.go [hunk 2]
```diff
@@ -27,3 +27,51 @@ func (r *Repository) CreatePet(ctx context.Context, p *gostreamv1.Pet) (uint64,
...
+	result := make([]*gostreamv1.Pet, 0)
+	for rows.Next() {
+		var u gostreamv1.Pet
+		if err = rows.Scan(&u.Id, &u.Kind, &u.Name, &u.Age); err != nil {
+			fmt.Println(err)
+		}
+
+		result = append(result, &u)
+	}
+	return result, err
```
Comment (lordvidex) [inline internal/db/pg/pets.go:55]: check if rows had error after scanning before returning results 

```go
if err = rows.Err(); err != nil {
 return nil, err
}
```

File: internal/db/pg/pets.go [hunk 3]
```diff
@@ -27,3 +27,51 @@ func (r *Repository) CreatePet(ctx context.Context, p *gostreamv1.Pet) (uint64,
...
+}
+
+// UpdatePet ...
+func (r *Repository) UpdatePet(ctx context.Context, p *gostreamv1.Pet) error {
+	id := p.GetId()
+	q := sq.Update("pets").Where("id = ?", id).
+		PlaceholderFormat(sq.Dollar)
+	q = q.Set("kind", p.GetKind())
+	q = q.Set("name", p.GetName())
+	q = q.Set("age", p.GetAge())
```
Comment (lordvidex) [inline internal/db/pg/pets.go:65]: Not so important but you can chain the queries like the following:

```go
q := sq.Update("pets").Where("id = ?", id).
		PlaceholderFormat(sq.Dollar).
                 Set("kind", p.GetKind()).
                 Set("name", p.GetName()).
	        Set("age", p.GetAge())
```

**Review by lordvidex — CHANGES_REQUESTED** [review]
I added some review comments

File: internal/db/pg/pets.go (hunk 3 above)
Comment (Joshdike) [reply internal/db/pg/pets.go]: done

File: internal/db/pg/pets.go (hunk 2 above)
Comment (Joshdike) [reply internal/db/pg/pets.go]: Done

**Comment by lordvidex** [comment]
I tried to resolve conflicts with the github interface but it's risky because there are conflicts in generated `pb.go` files and we might push breaking changes to main. It is better if you merge main into your current branch, rerun the generated code chunks and repush a fix commit.

**Pull request merged**